### PR TITLE
Minor updates (annotations and best-practices)

### PR DESCRIPTION
**Pull request by Theaxiom**
This is me splitting up a larger PR, please see #10532 for details.

### CODE REVIEW

File: src/Database/Schema/SqlserverSchema.php [hunk 1]
```diff
@@ -348,11 +348,12 @@ public function columnSql(TableSchema $schema, $name)
             $out .= $typeMap[$data['type']];
         }
 
-        if ($data['type'] === 'integer' || $data['type'] === 'biginteger') {
-            if ([$name] === $schema->primaryKey() || $data['autoIncrement'] === true) {
-                unset($data['null'], $data['default']);
-                $out .= ' IDENTITY(1, 1)';
-            }
+        if (
```
Comment (Theaxiom) [inline src/Database/Schema/SqlserverSchema.php]: Ahh, forgot to cherry-pick the commit that fixed that. Pulled it into the PR now.

**Comment by Theaxiom** [comment]
Looks like I have to fix `src/Database/Schema/Sqlserverschema.php`

File: src/Database/Schema/SqlserverSchema.php [hunk 2]
```diff
@@ -348,7 +348,7 @@ public function columnSql(TableSchema $schema, $name)
             $out .= $typeMap[$data['type']];
         }
 
-        if ($data['type'] === 'integer' || $data['type'] === 'biginteger') {
+        if ($data['type'] === 'integer' || ($data['type'] === 'biginteger') {
```
Comment (Theaxiom) [inline src/Database/Schema/SqlserverSchema.php]: Fixed.

File: src/Database/Dialect/SqliteDialectTrait.php [hunk 3]
```diff
@@ -120,7 +120,7 @@ protected function _transformFunctionExpression(FunctionExpression $expression)
                     ->setConjunction(' ,')
                     ->iterateParts(function ($p, $key) {
                         if ($key === 0) {
-                            $value = rtrim(strtolower($p), 's');
+                            $value = strtolower(rtrim($p, 's'));
```
Comment (lorenzo) [inline src/Database/Dialect/SqliteDialectTrait.php]: `rtrim('THINGS', 's')` is not the same as `rtrim('things', 's')`

File: src/ORM/Association.php [hunk 4]
```diff
@@ -1133,10 +1133,8 @@ protected function _appendFields($query, $surrogate, $options)
         $target = $this->_targetTable;
         $autoFields = $surrogate->isAutoFieldsEnabled();
 
-        if (empty($fields) && !$autoFields) {
-            if ($options['includeFields'] && ($fields === null || $fields !== false)) {
-                $fields = $target->getSchema()->columns();
-            }
+        if (empty($fields) && !$autoFields && $options['includeFields'] && ($fields === null || $fields !== false)) {
```
Comment (lorenzo) [inline src/ORM/Association.php]: This is very difficult to read. Why change it?

File: src/Database/Dialect/SqliteDialectTrait.php (hunk 3 above)
Comment (Theaxiom) [inline src/Database/Dialect/SqliteDialectTrait.php]: You are indeed correct.

File: src/Event/Decorator/ConditionDecorator.php [hunk 5]
```diff
@@ -63,11 +63,7 @@ public function canTrigger(Event $event)
     protected function _evaluateCondition($condition, Event $event)
     {
         if (!isset($this->_options[$condition])) {
-            if ($condition === 'unless') {
-                return false;
-            }
-
-            return true;
+            return !($condition === 'unless');
```
Comment (inoas) [inline src/Event/Decorator/ConditionDecorator.php]: The code before may be slightly more verbose but a lot better to read.

File: src/Filesystem/File.php [hunk 6]
```diff
@@ -106,13 +106,8 @@ public function __destruct()
...
     {
         $dir = $this->Folder->pwd();
-        if (is_dir($dir) && is_writable($dir) && !$this->exists()) {
-            if (touch($this->path)) {
-                return true;
-            }
-        }
 
-        return false;
+        return is_dir($dir) && is_writable($dir) && !$this->exists() && touch($this->path);
```
Comment (inoas) [inline src/Filesystem/File.php]: Worse to read

File: src/Filesystem/Folder.php [hunk 7]
```diff
@@ -631,20 +631,18 @@ public function create($pathname, $mode = false)
...
-                $this->_errors[] = sprintf('%s NOT created', $pathname);
+                $this->_messages[] = sprintf('%s created', $pathname);
 
-                return false;
+                return true;
             }
+            umask($old);
+            $this->_errors[] = sprintf('%s NOT created', $pathname);
+
+            return false;
```
Comment (inoas) [inline src/Filesystem/Folder.php]: Existing bail early is better practice.
Also such code changes unless accompanied by a specific unit test can introduce silent bugs

File: src/Http/Response.php [hunk 8]
```diff
@@ -1736,7 +1736,7 @@ protected function _getUTCDate($time = null)
         } else {
             $result = new DateTime($time);
         }
-        $result->setTimeZone(new DateTimeZone('UTC'));
+        $result->setTimezone(new DateTimeZone('UTC'));
```
Comment (inoas) [inline src/Http/Response.php]: Is the method called setTimeZone?

File: src/Database/Driver/PDODriverTrait.php [hunk 9]
```diff
@@ -155,7 +155,7 @@ public function rollbackTransaction()
             return false;
         }
 
-        return $this->_connection->rollback();
+        return $this->_connection->rollBack();
```
Comment (inoas) [inline src/Database/Driver/PDODriverTrait.php]: Is the method called rollBack?

File: src/Validation/ValidationRule.php [hunk 10]
```diff
@@ -169,7 +169,7 @@ protected function _skip($context)
 
         $newRecord = $context['newRecord'];
         if (!empty($this->_on)) {
-            if ($this->_on === 'create' && !$newRecord || $this->_on === 'update' && $newRecord) {
+            if (($this->_on === 'create' && !$newRecord) || ($this->_on === 'update' && $newRecord)) {
```
Comment (inoas) [inline src/Validation/ValidationRule.php]: I think this is okay to make comparison operator precedence explict.

Comment (inoas) [inline src/Validation/ValidationRule.php]: Aka was okay before

**Review by inoas — CHANGES_REQUESTED** [review]

File: src/Validation/ValidationRule.php (hunk 10 above)
Comment (dereuromark) [inline src/Validation/ValidationRule.php]: Being a bit more explicit here doesnt hurt.

File: src/Http/Response.php (hunk 8 above)
Comment (dereuromark) [inline src/Http/Response.php]: setTimezone() is the method. So :+1: here

File: src/Database/Driver/PDODriverTrait.php (hunk 9 above)
Comment (dereuromark) [inline src/Database/Driver/PDODriverTrait.php]: rollback() is the method name, so :-1:  here.

File: src/Filesystem/File.php [hunk 11]
```diff
@@ -106,8 +106,10 @@ public function __destruct()
     public function create()
     {
         $dir = $this->Folder->pwd();
+
         if (is_dir($dir) && is_writable($dir) && !$this->exists()) {
             if (touch($this->path)) {
+
```
Comment (dereuromark) [inline src/Filesystem/File.php]: Why this newline? Shouldnt be there IMO.

File: src/Template/Element/plugin_class_error.ctp [hunk 12]
```diff
@@ -29,5 +29,3 @@ else:
         '<a href="http://book.cakephp.org/3.0/en/plugins.html#autoloading-plugin-classes">Plugins - autoloading plugin classes</a>'
     );
 endif;
-
-?>
```
Comment (dereuromark) [inline src/Template/Element/plugin_class_error.ctp]: Templates should keep the closing tags - as they are mainly HTML, and there might be some more HTML that wants to be added below at some point.

File: src/Filesystem/File.php (hunk 11 above)
Comment (Theaxiom) [inline src/Filesystem/File.php]: New line before return, CS rule.

File: src/Template/Element/plugin_class_error.ctp (hunk 12 above)
Comment (Theaxiom) [inline src/Template/Element/plugin_class_error.ctp]: Then I would say update all the templates to include closing PHP tags.

Comment (Theaxiom) [inline src/Template/Element/plugin_class_error.ctp]: nvm, I just checked, the others don't require it

Comment (markstory) [inline src/Template/Element/plugin_class_error.ctp]: Not sure templates need a closing `?>` if they end in a PHP block.

File: src/Filesystem/File.php (hunk 11 above)
Comment (dereuromark) [inline src/Filesystem/File.php]: No, never as a single rested line.

Comment (Theaxiom) [inline src/Filesystem/File.php]: Ok, I will fix it. Thanks. 👍

File: src/Template/Element/plugin_class_error.ctp (hunk 12 above)
Comment (dereuromark) [inline src/Template/Element/plugin_class_error.ctp]: It makes it easier to add HTML below - without the hasle of having to add the closing tag again as well.

Comment (Theaxiom) [inline src/Template/Element/plugin_class_error.ctp]: I think I am inclined to agree with @dereuromark

**Comment by Theaxiom** [comment]
Is this PR ready? Please let me know if anything else needs to be addressed. 👍

**Comment by dereuromark** [comment]
Looks like this can be merged.

**Pull request merged**